File: testio/src/main/java/com/example/testio/injection/AppComponent.java
package com.example.testio.injection;

import android.content.Context;
import com.example.testio.App;
import com.example.testio.api.TestioApi;
import com.example.testio.helpers.TokenStorage;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = { AppModule.class })
public interface AppComponent {
  Context getAppContext();

  App getApp();

  TestioApi getTestioApi();

  TokenStorage getTokenStorage();
}

File: app/src/main/java/com/teso/net/di/AppComponent.kt
package com.teso.net.di

import com.teso.net.ui.vm.LoadingFragmentVM
import com.teso.net.ui.vm.LoginFragmentVM
import com.teso.net.ui.vm.MainActivityVM
import com.teso.net.ui.vm.ServerListVM
import dagger.Component
import javax.inject.Singleton


@Singleton
@Component(modules = [(AppModule::class)])
interface AppComponent {

    fun inject(siteFragmentVM: ServerListVM)

    fun inject(loginFragmentVM: LoginFragmentVM)

    fun inject(loadingFragmentVM: LoadingFragmentVM)

    fun inject(mainActivityVM: MainActivityVM)
}